chalk: init at 1.29.4

curranosaurus opened the pull request:
I'm adding myself as a maintainer, in order to add a new package in a followup PR.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

curranosaurus commented:
@9999years and/or @lf-, would you be willing to review this? I've used my work email, not my personal email, since I don't think I will be maintaining anything from a hobbyist point of view. I'm new to being a maintainer of anything, so let me know if that sounds reasonable or not!

dotlambda reviewed (commented):
No need to do this in a separate PR. You can just include the commit in the PR in which you add a package.
Anyway, the commit message and PR title should be `maintainers: add curran`.

lf- reviewed (commented):
please add yourself to the teams file on the Mercury team if you think it's appropriate :)

https://github.com/NixOS/nixpkgs/blob/master/maintainers/team-list.nix

curranosaurus commented:
Okay cool, I added myself to the team list and changed the PR name and commit message. It sounds like it's fine to have this PR both add me as a maintainer, and add the package that I am looking to add, so I will follow shortly up with a commit to add that package shortly.

HeitorAugustoLN commented:
Add yourself to the team in a separate commit with the following message:
`maintainers/team-list: add curran to mercury`

curranosaurus committed:
maintainers: add curran

curranosaurus committed:
maintainers/team-list: add curran to mercury

dotlambda commented on pkgs/by-name/ch/chalk/package.nix:
```diff
@@ -0,0 +1,59 @@
+  chalkUrl = "https://api.chalk.ai/v1/install/${chalkPathPieces}/v${chalkVersion}";
+in
+stdenv.mkDerivation {
+  name = "chalk";
```
```suggestion
  pname = "chalk";
```

dotlambda commented on pkgs/by-name/ch/chalk/package.nix:
```diff
@@ -0,0 +1,59 @@
+  dontUnpack = true;
+
+  outputs = [
+    "out"
```
redundant

9999years replied:
Yeah, you can remove the whole `outputs` declaration here.

dotlambda commented on pkgs/by-name/ch/chalk/package.nix:
```diff
@@ -0,0 +1,59 @@
+  ];
+
+  installPhase = ''
+    mkdir -p $out/bin
```
```suggestion
    runHook preInstall

    mkdir -p $out/bin
```
and postInstall

9999years reviewed (commented):
Great start! Just fix up the nits that @dotlambda left and I think we'll be good to merge this.

curranosaurus committed:
chalk: init at 1.29.4

curranosaurus commented:
Hmm, I'm getting a build error now. I might have used the preInstall hook wrong? Going to try to debug.

curranosaurus commented:
Ah I deleted the `mkdir` by mistake. Building works locally again on my machine now. Thanks for the corrections, @dotlambda!

dotlambda commented on pkgs/by-name/ch/chalk/package.nix:
```diff
@@ -0,0 +1,57 @@
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    cp $src $out/bin/chalk
+    chmod +x $out/bin/chalk
```
```suggestion
    install -tm 555 $src $out/bin/chalk
```

curranosaurus commented:
Thanks everyone!

dotlambda commented on pkgs/by-name/ch/chalk/package.nix:
```diff
@@ -0,0 +1,55 @@
+    "aarch64-linux" = "sha256-XHaCLxVJbXjPILczDGWLFqP0q/nBO5O2A9lghkvM474=";
+    "x86_64-linux" = "sha256-hlNljLJm+m7l+Djni+ATKyWKSGKSDP0YN3CuJ4fXmWg=";
+  };
+  chalkHash = chalkHashByNixSystem."${stdenv.system}";
```
https://github.com/NixOS/nixpkgs/issues/27069
